lib/std/tempfiles.nim

```diff
@@ -80,21 +100,30 @@ proc mkstemp*(prefix, suffix: string, dir = ""): tuple[fd: File, name: string] =
   result.name.setLen(dir.len + prefix.len + nimTempPathLength + suffix.len + 2)
 
   for i in 0 ..< maxRetry:
-    result.name = joinPath(dir, prefix & randomPathName(nimTempPathLength) & suffix)
+    result.name = dir / prefix & randomPathName(nimTempPathLength) & suffix
```

Comment:
```suggestion
    result.name = dir / (prefix & randomPathName(nimTempPathLength) & suffix)
```

code was buggy when prefix.len == 0, try it!

Reply:
fixed